contracts/epochs/FlowEpoch.cdc

```diff
@@ -361,12 +384,25 @@ pub contract FlowEpoch {
         }
     }
 
-    /// Pays rewards, moves staking tokens between buckets,
-    /// and starts the new epoch staking auction
-    access(account) fun startNewEpoch() {
+    /// Pays rewards to staked nodes and delegators,
+    /// Calculates a new token payout, and sets the new payout
+    access(account) fun payRewards(_ newPayout: UFix64?) {
```

Comment:
With this change, it seems like we now provide the reward amount for _next_ epoch when paying out rewards for the _current_ epoch. Is that right? I'm a bit worried about the indirection and the change (I believe) from the current functionality for a manual process.

Reply:
yes, you are correct. The current functionality would stay the same until we decide to change it, because the line is commented out. How else would you recommend we do it when it is fully automated? To me this is the only logical place do calculate the rewards, because the rewards depend on how many tokens are actually rewarded in the current epoch, which could change because of arithmetic precision loss when calculating it for each staker. And we shouldn't calculate it  after the epoch starts, because the stakers should know, at the beginning of the epoch, the number of tokens that are going to get paid.

Reply:
> The current functionality would stay the same until we decide to change it

This might be the part I'm missing. I assumed we would would use this function for the manual payout as well. But I guess we would do something like:

```
stakingAdmin.setEpochTokenPayout(currentEpochPayout)
stakingAdmin.payRewards()
```

Reply:
We would use this function. This function behaves in exactly the same way that we do it now. We pay rewards, then we set the new payout for the next epoch. It just happens in the function in stead of in separate function calls